OpenProjects.html

```diff
@@ -466,6 +469,59 @@
   </p>
 </div>
 
+<!-- *********************************************************************** -->
+<div class="www_subsubsection">
+  <a name="clang-plugins-windows">Support clang plugins on Windows</a>
+</div>
+<!-- *********************************************************************** -->
+
+<div class="www_text">
+  <p><b>Description of the project:</b>
+    The Clang compiler is part of the LLVM compiler infrastructure and supports
+    various languages such as C, C++, ObjC and ObjC++. The design of LLVM and
+    Clang allows the compiler to be extended with plugins[1]. A plugin makes it
+    possible to run extra user defined actions during a compilation. Plugins
+    are supported on unix and darwin but not on windows due to some specifics of
+    the windows platform.
+    <br /> <br />
+    The general work that needs to be done is to annotate all the public API
+    surface with __declspec(dllexport) and similar if they are being built as a
+    dynamically linked library (and nothing if being built statically). When the
+    library is being consumed the attribute on the same declaration must be
+    __declspec(dllimport) or the same spelling for the GNU version for ELF and
+    MachO. Note that the attribute changes between the implementation and
+    consumer and between static and dynamic linking. This is normally controlled
+    by expansion to a macro.
```

Comment:
```suggestion
    This project would expose the participant to a broad cross section of the LLVM codebase. It involves exploring the API surface, classifying the interfaces as being public or private, and annotating that information to the API declarations. It would also expose the participant to details and differences of different platforms as this work is cross-platform (Windows, Linux, Darwin, BSD, etc). The resulting changes would improve LLVM on Linux and Windows while enabling new functionality on Windows.
```